Files/Helpers/AdaptiveLayoutHelpers.cs

```diff
@@ -84,47 +88,69 @@ public static bool PredictLayoutMode(FolderSettingsViewModel folderSettings, Ite
-                    { // Most of items are images/videos
-                        folderSettings.ToggleLayoutModeTiles.Execute(false);
+                mediaCount = filesystemViewModel.FilesAndFolders.Where((item) =>
+                {
+                    return !string.IsNullOrEmpty(item.FileExtension) && MediaPreviewViewModel.Extensions.Any((ext) => item.FileExtension.Equals(ext, StringComparison.OrdinalIgnoreCase));
+                }).Count();
+                imagesCount = filesystemViewModel.FilesAndFolders.Where((item) =>
+                {
+                    return !string.IsNullOrEmpty(item.FileExtension) && ImagePreviewViewModel.Extensions.Any((ext) => item.FileExtension.Equals(ext, StringComparison.OrdinalIgnoreCase));
+                }).Count();
+                foldersCount = filesystemViewModel.FilesAndFolders.Where((item) => item.PrimaryItemAttribute == StorageItemTypes.Folder).Count();
+                miscFilesCount = allItemsCount - (mediaCount + imagesCount + foldersCount);
+
+                mediaPercentage = (float)((float)mediaCount / (float)allItemsCount) * 100.0f;
+                imagesPercentage = (float)((float)imagesCount / (float)allItemsCount) * 100.0f;
+                foldersPercentage = (float)((float)foldersCount / (float)allItemsCount) * 100.0f;
+                miscFilesPercentage = (float)((float)miscFilesCount / (float)allItemsCount) * 100.0f;
+
+                // Decide layout mode
+                
+                // Mostly files + folders, lesser media and image files | Mostly folders
+                if ((foldersPercentage + miscFilesPercentage) > 80.0f)
+                {
+                    if (allItemsCount > 6)
+                    {
+                        layoutDetails();
                     }
                     else
                     {
-                        folderSettings.ToggleLayoutModeDetailsView.Execute(false);
+                        layoutTiles();
                     }
                 }
-                else
-                { // There are only files
-                    if (imagesAndVideosCount == filesystemViewModel.FilesAndFolders.Count)
-                    { // Only images/videos
-                        folderSettings.ToggleLayoutModeGridView.Execute(folderSettings.GridViewSize);
-                    }
-                    else if (otherFilesCount < 20)
-                    { // Most of files are images/videos
-                        folderSettings.ToggleLayoutModeTiles.Execute(false);
+                // Mostly images, probably an images folder
+                else if (imagesPercentage > 85.0f || (imagesPercentage > 60.0f && (mediaPercentage + miscFilesPercentage + foldersPercentage) > 25.0f && (miscFilesPercentage + foldersPercentage) < 15.0f))
+                {
```

Comment:
@d2dyno1 How comfortable would you be with making the threshold percentages constant variables? I just prefer to reduce the amount of "magic numbers" if possible.

Reply:
Sure, can you give some suggestions how to name them? 💡

Reply:
Some of the weird names we had in Files demonstrate I'm probably the wrong person for that job 😆

@yaichenbaum We need some variable names for these threshold values. 
How about the following:
- `LargeAdaptiveThreshold` for 85.0f
- `MediumAdaptiveThreshold` for 60.0f
- `SmallAdaptiveThreshold` for 25.0f
- `ExtraSmallAdaptiveThreshold` for 15.0f

Reply:
If these sound weird, then we can drop this suggestion altogether, but it would reduce duplicated use of magic float values.

Reply:
How about naming them with different confidence levels?

Reply:
@duke7553 @yaichenbaum check out `9af6e4d`!

Reply:
Yikes

Reply:
> Yikes

![image](https://user-images.githubusercontent.com/53011783/120939677-6bb3fe00-c719-11eb-9091-5efa6b32ce16.png)

Reply:
Any ideas to improve it?

Reply:
@duke7553 had a good idea in his comment above.